app/controllers/concerns/course_basic_detail_concern.rb

```diff
@@ -89,18 +104,40 @@ def build_previous_course_creation_params
   end
 
   def next_step(current_step:)
-    if current_step == :outcome
-      new_provider_recruitment_cycle_courses_entry_requirements_path(
-        params[:provider_code],
-        params[:recruitment_cycle_year],
-        course: course_params,
-      )
-    elsif current_step == :entry_requirements
-      new_provider_recruitment_cycle_courses_outcome_path(
-        params[:provider_code],
-        params[:recruitment_cycle_year],
-        course: course_params,
-      )
+    next_step = NextCourseCreationStepService.new.execute(current_step: current_step)
+    next_page = course_creation_path_for(next_step)
+
+    if next_page.nil?
+      raise "No path defined for next step: #{next_step}"
+    end
+
+    next_page
+  end
+
+  def course_creation_path_for(page)
+    path_params = { course: course_params }
+
+    case page
+    when :apprenticeship
+      new_provider_recruitment_cycle_courses_apprenticeship_path(path_params)
+    # Currently the location page isnt built - so we skip that step
+    # and go to the entry_requirements for now
+    when :location
+      new_provider_recruitment_cycle_courses_entry_requirements_path(path_params)
+    when :entry_requirements
+      new_provider_recruitment_cycle_courses_entry_requirements_path(path_params)
+    when :outcome
+      new_provider_recruitment_cycle_courses_outcome_path(path_params)
+    when :full_or_part_time
```

Comment:
just fussy... 

can we not just called this `study_mode`

Reply:
I tried to keep it in line with what the URLs were as opposed to the name of the field (e.g. Apprenticeship over Program Type)

Reply:
I agree, we may change `study_mode` in the backend, and we shouldn't have to change this as a result, I'd say.